### DISCUSSION TITLE
Is there a way to set options (key-value) of dropdowns programmatically?

### DISCUSSION THREAD

**Discussion by SangaRavi** [writer/writer-framework#46] [General]
Currently, I have to set the value of drop downs on frontend editor before starting the app. I would like to set it dynamically based on another control's chosen value. Is this possible?

**Comment by ramedina86** [comment]
Hi, thanks for trying Streamsync. Yes, it's possible.

You can store the options in state. Then, rather than "Static List", select "JSON" when specifying the options. There, include a reference to state such as `@{cities}`. The `cities` state element should contain a dictionary. It can be an empty dictionary when initialising.

You can alter the `cities` value and this will change the options available. To dynamically change it based on another control, you can use an event handler.

For example, to update `cities` based on country, you'd link this event handler in a "Country" control.

```py
def update_cities(state, payload):
    if payload == "ar":
        state["cities"] = {
            "ba": "Buenos Aires",
            "cb": "Cordoba"
        }
    elif payload == "uk":
        state["cities"] = {
            "ln": "London",
            "br": "Bristol"
        }
```

Hope this helps!
